doc/api/stream.md

```diff
@@ -883,83 +1035,109 @@ implement Readable streams in your programs.
 
-#### readable.push(chunk[, encoding])
+The `readable.\_read()` method is prefixed with an underscore because it is
+internal to the class that defines it, and should never be called directly by
+user programs.
 
+#### readable.push(chunk[, encoding])
 
 * `chunk` {Buffer|Null|String} Chunk of data to push into the read queue
 * `encoding` {String} Encoding of String chunks.  Must be a valid
   Buffer encoding, such as `'utf8'` or `'ascii'`
 * return {Boolean} Whether or not more pushes should be performed
 
-Note: **This method should be called by Readable implementors, NOT
-by consumers of Readable streams.**
+*Note*: The `readable.push()` method is intended be called only by Readable
+implementors, and **not** by consumers of Readable streams.
 
-If a value other than null is passed, The `push()` method adds a chunk of data
-into the queue for subsequent stream processors to consume. If `null` is
-passed, it signals the end of the stream (EOF), after which no more data
-can be written.
+When `chunk` is a `Buffer` or `string`, the `chunk` of data will be added to the
+internal queue for users of the stream to consume. Passing `chunk` as `null` 
+signals the end of the stream (EOF), after which no more data can be written.
 
-The data added with `push()` can be pulled out by calling the
-[`stream.read()`][stream-read] method when the [`'readable'`][] event fires.
+When the Readable is operating in paused mode, the data added with
+`readable.push()` can be read out by calling the [`stream.read()`][stream-read]
+method when the [`'readable'`][] event fires.
 
-This API is designed to be as flexible as possible. For example,
-you may be wrapping a lower-level source which has some sort of
-pause/resume mechanism, and a data callback. In those cases, you
-could wrap the low-level source object by doing something like this:
+When the Readable is operating in flowing mode, the data added with
+`readable.push()` will be delivered by emitting a `'data'` event.
+
+The `readable.push()` method is designed to be as flexible as possible. For
+example, when wrapping a lower-level source that provides some form of
+pause/resume mechanism, and a data callback, the low-level source can be wrapped
+by the custom Readable instance as illustrated in the following example:
 
 ```js
 // source is an object with readStop() and readStart() methods,
 // and an `ondata` member that gets called when it has data, and
 // an `onend` member that gets called when the data is over.
 
-
 class SourceWrapper extends Readable {
   constructor(options) {
     super(options);
 
     this._source = getLowlevelSourceObject();
 
-    // Every time there's data, we push it into the internal buffer.
+    // Every time there's data, push it into the internal buffer.
     this._source.ondata = (chunk) => {
-      // if push() returns false, then we need to stop reading from source
+      // if push() returns false, then stop reading from source
```

Comment:
A few more questions: What happens if you call `.push` outside of `_read()`? What happens if you `readable.emit('data', mychunk)`?